hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,381 @@
+          unReplicatedSizeCountMap.put(
+              getUnReplicatedSizeKeyFromTable(tableName), details.getMiddle());
+          replicatedSizeCountMap.put(
+              getReplicatedSizeKeyFromTable(tableName), details.getRight());
+        } else {
+          long count = getCount(table.iterator());
+          objectCountMap.put(getTableCountKeyFromTable(tableName), count);
+        }
+      } catch (IOException ioEx) {
+        LOG.error("Unable to populate Table Count in Recon DB.", ioEx);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+
+    // Write count data to DB if the sizeCountMap is not empty
+        if (!objectCountMap.isEmpty()) {
+          writeDataToDB(objectCountMap);
+        }
+
+    // Write unreplicated size data to DB if the sizeCountMap is not empty
+        if (!unReplicatedSizeCountMap.isEmpty()) {
+          writeDataToDB(unReplicatedSizeCountMap);
+        }
+
+    // Write replicated size data to DB if the sizeCountMap is not empty
+        if (!replicatedSizeCountMap.isEmpty()) {
+          writeDataToDB(replicatedSizeCountMap);
+        }
+
+    LOG.info("Completed a 'reprocess' run of OmTableInsightTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  /**
+   * Returns a pair with the total count of records (left), total unReplicated
+   * size (middle), and total replicated size (right) in the given table.
+   * Increments count for each record and adds the dataSize if a record's value
+   * is an instance of OmKeyInfo. If the table is null, returns (0, 0, 0).
+   *
+   * @param table The table from which to get the count and data size.
+   * @return Three Long values representing the count, unReplicated size,
+   *         and replicated size.
+   * @throws IOException If an I/O error occurs during the table iteration.
+   */
+  private Triple<Long, Long, Long> getTableSizeAndCount(Table table)
+      throws IOException {
+    long count = 0;
+    long unReplicatedSize = 0;
+    long replicatedSize = 0;
+
+    TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator =
+        table.iterator();
+    while (iterator.hasNext()) {
+      Table.KeyValue<String, ?> kv = iterator.next();
+
+      if (kv != null && kv.getValue() != null) {
+        if (kv.getValue() instanceof OmKeyInfo) {
+          OmKeyInfo omKeyInfo = (OmKeyInfo) kv.getValue();
+          unReplicatedSize += omKeyInfo.getDataSize();
+          replicatedSize += omKeyInfo.getReplicatedSize();
+        }
+      }
+      count++;  // Increment count for each row
+    }
+    return Triple.of(count, unReplicatedSize, replicatedSize);
+  }
+
+  /**
+   * Returns a collection of table names that require data size calculation.
+   */
+  public Collection<String> getTablesToCalculateSize() {
+    List<String> taskTables = new ArrayList<>();
+    taskTables.add(OPEN_KEY_TABLE);
+    taskTables.add(OPEN_FILE_TABLE);
+    return taskTables;
+  }
+
+  private long getCount(Iterator iterator) {
```

Comment:
No need of this method as we already have built-in method Iterators.size().

Reply:
Thanks for the reference @ashishkumar50 I have made the required changes!